docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,216 @@
+# Authentication and Authorization
+
+## Introduction
+Currently, communication between the Compass and both Runtimes and Applications are not secured. We need to provide some security possibilities.
+We want to secure the Compass using ORY's Hydra and OathKeeper. There would be three ways of authentication:
+ - OAuth 2.0 
+ - Client Certificates (mTLS)
+ - JWT token issued by Dex
+
+ There would be three possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. We also need to implement additional supporting components to make our scenarios valid.
+
+## Architecture
+
+The following diagram represents the architecture of the security in Compass:
+
+![](./assets/security-architecture.svg)
+
+### Tenant Mapping Handler
+
+It is an OathKeeper [hydrator](https://github.com/ory/docs/blob/525608c65694539384b785355d293bc0ad00da27/docs/oathkeeper/pipeline/mutator.md#hydrator) handler responsible for mapping authentication session data to tenant. It could be built into Director itself, as it will also use the same database. It would be implemented as a separate endpoint, such as `/tenant-mapping`.
+
+To unify the approach for mapping, we could introduce `authorization_id`, widely used by multiple authentication flows.
```

Comment:
> maybe even mention here about system_auths table, and not at the end of that chapter

It will be hard to do it at the beginning, because it's a technical detail. Need for `system_auths` table exists because of the `authorization_id`, so it's better to introduce this concept first.

Reply:
> how we map user to tenant by system_auths from: "JWT token issued by identity service" chapter "point 3".

It is described in the `JWT token issued by identity service` flow.

Reply:
1) what's about those could/should?
2) fine
3) does it means user flows, uses system_auths or not? In case of user and local/prodcution what is exactly in system_auths.

Reply:
3: On local system we use static ConfigMap, and for production system we must have tenant and scopes in token's payload. So, `system_auths` table is not used in JWT token from identity service flow.